src/lib/DiscordWorker.ts

```diff
@@ -26,7 +26,7 @@ export class DiscordJob extends Job {
   ) => Promise<
     OmitPartialGroupDMChannel<Message<true>> | Message<true> | undefined
   >
-  setThreadName: (name: string) => Promise<TextChannel>
```

Comment:
if we just avoid calling this function when there is no thread, then we don't need the undefined return type here (see comment below)